engine/Controller/SoundEngine.h

```diff
@@ -6,111 +6,37 @@
 #include <glm/glm.hpp>
 #pragma comment(lib, "irrKlang.lib")
 
-/**
-*	@class SoundEngine
-*	@brief contains basic function play sounds within a 3D space
-*	@version 1
-*	@author Dylan Burgess
-*	@date 07/06/2023
-*/
-
 class SoundEngine
 {
 public:
-	/**
-	*	@brief default constructor
-	*/
 	SoundEngine();
 
-	/**
-	*	@brief gets the sound engine singleton
-	*	@return SoundEngine
-	*/
 	static SoundEngine& Get();
-	/**
-	*	@brief adds sounds to the sound map
-	*   @param name - the key for the map
-	*   @param filePath - location of the sound
-	*	@return void
-	*/
-	void addSound(std::string name, std::string filePath);
 
-	/**
-	*	@brief plays a sounds from the sound map
-	*   @param name - the sound to play
-	*   @param pos - position of the sound to play
-	*	@return void
-	*/
-	void playSound(std::string name, glm::vec3 pos);
+	void playSoundFromFile(std::string audioFilePath);
 
-	/**
-	*	@brief adds music to the sound engine
-	*	@param filePath - location of the music
-	*	@return void
-	*/
-	void addMusic(std::string filePath);
-	/**
-	*	@brief set the volume of the sound engine
-	*	@param volume the perctane of volume 0-1
-	*	@return void
-	*/
-	void setVolume(float volume);
+	void playSound(std::string audioName);
 
-	/**
-	*	@brief set the volume of the music engine
-	*	@param volume the perctane of volume 0-1
-	*	@return void
-	*/
-	void setMusicVolume(float volume);
+	void playLoop(std::string audioName);
 
-	/**
-	*	@brief set the position of the sound listener
-	*	@param pos - position of the listener
-	*	@return void
-	*/
-	void setListenerPos(glm::vec3 pos);
-	/**
-	*	@brief set the position of the music player
-	*	@param pos - position of the music player
-	*	@return void
-	*/
-	void setMusicPos(glm::vec3 pos);
+	void addSound(std::string audioName, std::string audioFilePath);
 
-	/**
-	*	@brief turns on or off the music
-	*	@param state - true for on / false for off
-	*	@return void
-	*/
-	void toggleMusic(bool state);
-	/**
-	*	@brief memory cleanup
-	*	@return void
-	*/
-	void drop();
+	void playSoundAtPosition(std::string audioName, glm::vec3 pos);
 
-	/**
-	*	@brief plays a looped sound
-	*	@param name - the sound to play
-	*	@param pos - position of the sound to play
-	* 	@param volume the perctane of volume 0-1
-	*	@return void
-	*/
-	void playSoundLoop(std::string name, glm::vec3 pos, float vol);
+	irrklang::ISound* playDynamicSound(std::string audioName, glm::vec3 pos);
 
+	void playLoopAtPosition(std::string audioName, glm::vec3 pos);
 
-	/**
-	*	@brief stops a looped sound
-	*	@param name - the sound to play
-	*	@return void
-	*/
-	void stopSoundLoop(std::string name);
+	void setUserPosition(glm::vec3 pos);
 
-private:
+	void setAudioPosition(irrklang::ISound* audio, glm::vec3 pos);
```

Comment:
Should not use irrkland specific datatype as a parameter. Were facading it so users shouldnt see any irrkland stuff.

Reply:
Yeah this function is still a WIP that I'd like to have but not necessary for milestone 1. Has been commented out for now.